File: app/src/main/java/com/mmutert/freshfreezer/data/converters/AmountUnitConverter.java
package com.mmutert.freshfreezer.data.converters;

import androidx.room.TypeConverter;

import com.mmutert.freshfreezer.data.AmountUnit;

public class AmountUnitConverter {

    @TypeConverter
    public static String fromAmountUnit(AmountUnit unit) {
        return unit.toString();
    }

    @TypeConverter
    public static AmountUnit toAmountUnit(String unit) {
        return AmountUnit.valueOf(unit);
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/data/converters/AmountUnitConverter.kt
package com.mmutert.freshfreezer.data.converters

import androidx.room.TypeConverter
import com.mmutert.freshfreezer.data.AmountUnit

object AmountUnitConverter {
    @JvmStatic
    @TypeConverter
    fun fromAmountUnit(unit: AmountUnit): String {
        return unit.toString()
    }

    @JvmStatic
    @TypeConverter
    fun toAmountUnit(unit: String?): AmountUnit {
        return AmountUnit.valueOf(unit!!)
    }
}